### ISSUE TITLE
[LifetimeSafety] Support C compound literals

### ISSUE THREAD

**Issue by NeKon69** [llvm/llvm-project#206347]
LifetimeSafety does not fully model C compound literals yet.

Minimal repro:

```c
int *bad(void) {
  int *p = &(int){12};
  return p; // should warn
}
```

https://godbolt.org/z/6oPvfr9bn

In C, compound literal has automatic storage duration, so this should behave roughly like returning the address of an unnamed local variable.

Expected: warn that the returned pointer refers to storage that does not live long enough.

This should remain OK however:

```c
int ok(void) {
  int *p = &(int){12};
  *p = 100;
  return *p;
}
```

**Comment by NeKon69** [comment]
This appears to be much harder than I initially anticipated as these are not modeled on the CFG level. So this might actually take a while..
